crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -41,187 +55,926 @@ const EXPORTS_MODULE_NAME: &str = "exports";
+    ///     first: string;
+    ///     second: u32;
+    ///     third: bool;
+    ///   }
+    ///
+    ///   g: func(input: g-request);
+    ///
+    ///   h: func(first: bool, second: string);
+    /// }
+    /// ```
+    ///
+    /// Under this setting, bindgen will not produce an error on any function, but will be slightly
+    /// inefficient as it will wrap `f` in a generated `ExamplesArgBundleFInvocation` struct (containing one member, `input`).
+    BundleArguments,
+}
+
+impl WitFunctionLatticeTranslationStrategy {
+    /// Translate a wit-bindgen generated trait function for use across the lattice
+    fn translate_import_fn_for_lattice(
+        &self,
+        wit_iface_name: WitInterfaceName,
+        trait_method: &TraitItemMethod,
+        struct_lookup: &StructLookup,
+        type_lookup: &TypeLookup,
+        bindgen_cfg: &ProviderBindgenConfig,
+    ) -> anyhow::Result<(WitInterfaceName, LatticeMethod)> {
+        // Calculate the lattice method name
+        let lattice_method_name = LitStr::new(
+            match (
+                bindgen_cfg.contract.as_str(),
+                &bindgen_cfg.wit_ns,
+                &bindgen_cfg.wit_pkg,
+            ) {
+                // To facilitate backwards compatibility with existing wasmcloud:keyvalue 
+                ("wasmcloud:keyvalue", _, _) => {
+                    format!(
+                        "KeyValue.{}",
+                        trait_method.sig.ident.to_string().to_upper_camel_case(),
+                    )
+                }
+                // Generic handling for other WIT worlds
+                (_, Some(ns), Some(pkg)) => format!(
+                    "{ns}:{pkg}/{}.{}",
+                    trait_method.sig.ident.to_string().to_upper_camel_case(),
+                    wit_iface_name.to_upper_camel_case()
+                ),
+                // Generally the only thing that can be missing here is either the WIT ns or pkg
+                _ => {
+                    bail!("missing/invalid WIT namespace or package. Consider explicitly specifying via the 'wit_namespace'/'wit_package' bindgen arguments")
+                }
+            }
+            .as_ref(),
+            Span::call_site(),
+        );
+
+        match self {
+            WitFunctionLatticeTranslationStrategy::Auto => match trait_method.sig.inputs.len() {
+                0 | 1 => Self::translate_import_fn_via_first_arg(
+                    wit_iface_name,
+                    lattice_method_name,
+                    trait_method,
+                    struct_lookup,
+                    type_lookup,
+                ),
+                _ => Self::translate_import_fn_via_bundled_args(
+                    wit_iface_name,
+                    lattice_method_name,
+                    trait_method,
+                    struct_lookup,
+                    type_lookup,
+                ),
+            },
+            WitFunctionLatticeTranslationStrategy::FirstArgument => {
```

Comment:
Since we can easily detect the arity of the function, I'm wondering when we would ever explicitly specify `FirstArgument` instead of `Auto`

Reply:
Ah so somehow Github hid this comment from me, but this was in there to facilitate some contracts that are inconsistent.

We can't control what's in every WIT contract, and some contracts might have a single all-in-one arg to a function but then other functions with different #s of args. I was thinking was that you'd probably *want* to force bundling to make it consistent... 

But this is something I could see just leaving at auto and then allowing mixing?

Reply:
Resolving for now -- we can always come back to this and remove it if it proves unnecessary.